runtimes/common/src/bonded_coins/mod.rs

```diff
@@ -0,0 +1,352 @@
+// KILT Blockchain – https://botlabs.org
+// Copyright (C) 2019-2024 BOTLabs GmbH
+
+// The KILT Blockchain is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The KILT Blockchain is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+// If you feel like getting in touch with us, you can do so at info@botlabs.org
+
+/// Wrapper struct for [UnionOf] to implement the [metadata::Inspect] trait,
+/// needed for the pallet_bonded_coins module.
+use frame_support::{
+	dispatch::DispatchResult,
+	traits::{
+		fungible::{self, UnionOf},
+		fungibles::{self, metadata::Inspect},
+		tokens::{
+			AssetId as AssetIdTraits, DepositConsequence, Fortitude, Precision, Preservation, Provenance,
+			WithdrawConsequence,
+		},
+		AccountTouch, Get,
+	},
+};
+use sp_runtime::{
+	traits::Convert,
+	DispatchError, Either,
+	Either::{Left, Right},
+};
+use sp_std::{marker::PhantomData, vec::Vec};
+use substrate_fixed::{
+	traits::Fixed,
+	types::{I75F53, U75F53},
+};
+
+pub mod hooks;
+
+pub mod runtime_api;
+
+/// The AssetId for bonded assets.
+pub type AssetId = u32;
+
+/// Fixed point number used for creating bonding curves.
+pub type FixedPointInput = U75F53;
+
+/// Fixed point number used for doing calculation steps in the bonding curves.
+pub type FixedPoint = I75F53;
+
+/// For a I75F53, the underlying type is a i128.
+pub type FixedPointUnderlyingType = <FixedPoint as Fixed>::Bits;
+
+/// Struct to implement the desired [Convert] trait needed for the
+/// [NativeAndForeignAssets] type.
+/// The generic type Target is used to determine the type of the asset id
+/// for the Either::Left variant.
+pub struct TargetFromLeft<Target>(PhantomData<Target>);
+
+/// Metadata trait for native asset.
+pub trait InspectMetadata<AssetId> {
```

Comment:
This trait actually belongs to `kilt_support`, and not `runtime_common`.

Reply:
Also, what's the need for `AssetId`? It's not used anywhere in the trait.

Reply:
Ah yeah. Good catch: [7af4e61](https://github.com/KILTprotocol/kilt-node/pull/818/commits/7af4e61675178db659f264da90c7a70061475c6a)